dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -304,13 +324,63 @@ public Service initializeService(String repositoryId) {
         service.setLastUpdated(new Date());
         service.setDescriptorType(DescriptorLanguage.SERVICE);
         service.setMode(WorkflowMode.SERVICE);
-        service.setDefaultWorkflowPath(".dockstore.yml");
+        service.setDefaultWorkflowPath("/.dockstore.yml");
+        return service;
+    }
+
+    /**
+     * Initialize service object for GitHub repository
+     * @param repositoryId Organization and repository (ex. dockstore/dockstore-ui2)
+     * @param subclass The subclass of the workflow (ex. docker-compose)
+     * @return Service
+     */
+    public Service initializeServiceFromGitHub(String repositoryId, String subclass) {
+        Service service = initializeService(repositoryId);
+        service.setMode(WorkflowMode.DOCKSTORE_YML);
+
+        // Validate subclass
+        if (subclass != null) {
+            DescriptorLanguageSubclass descriptorLanguageSubclass;
+            try {
+                descriptorLanguageSubclass = DescriptorLanguageSubclass.convertShortNameStringToEnum(subclass);
+            } catch (UnsupportedOperationException ex) {
+                String msg = "Subclass " + subclass + " is not a valid descriptor language subclass.";
+                LOG.info(msg);
+                throw new CustomWebApplicationException(msg, LAMBDA_FAILURE);
+            }
+            service.setDescriptorTypeSubclass(descriptorLanguageSubclass);
+        }
+
         return service;
     }
 
+    /**
+     * Initialize workflow object for GitHub repository
+     * @param repositoryId Organization and repository (ex. dockstore/dockstore-ui2)
+     * @param subclass Subclass of the workflow
+     * @param workflowName Name of the workflow
+     * @return Workflow
+     */
+    public BioWorkflow initializeWorkflowFromGitHub(String repositoryId, String subclass, String workflowName) {
+        BioWorkflow workflow = (BioWorkflow)initializeWorkflow(repositoryId, new BioWorkflow());
+        workflow.setWorkflowName(workflowName);
+        workflow.setMode(WorkflowMode.DOCKSTORE_YML);
+        if (Objects.equals(DescriptorLanguage.CWL.getLowerShortName(), subclass)) {
```

Comment:
Use https://github.com/dockstore/dockstore/blob/develop/dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguage.java#L146 or similar so that we don't need to touch this when we need to get Galaxy in here too.